flink/v1.18/flink/src/main/java/org/apache/iceberg/flink/source/reader/IcebergSourceSplitReader.java

```diff
@@ -46,6 +51,8 @@ class IcebergSourceSplitReader<T> implements SplitReader<RecordAndPosition<T>, I
   private final SerializableComparator<IcebergSourceSplit> splitComparator;
   private final int indexOfSubtask;
   private final Queue<IcebergSourceSplit> splits;
+  private final Set<String> pausedSplits;
+  private final AtomicBoolean wakeUp = new AtomicBoolean(false);
```

Comment:
nit: since we are using `synchronized` block, wondering what is the typical style: plain boolean vs AtomicBoolean?

Reply:
You can not synchronize on `boolean`, and you can not set `Boolean`. That is why I used `AtomicBoolean`.

Reply:
if it is non-final, we can set the value.. anyway, it is trivial. not worth further discussion